Mathlib/Topology/Algebra/Valued/WithVal.lean

```diff
@@ -124,6 +158,57 @@ abbrev Completion := UniformSpace.Completion (WithVal v)
 instance : Coe R v.Completion :=
   inferInstanceAs <| Coe (WithVal v) (UniformSpace.Completion (WithVal v))
 
+section Equivalence
+
+/-! The uniform isomorphism between `WithVal v` and `WithVal w` when `v` and `w` are
+equivalent. -/
+
+variable {R Γ₀ Γ₀' : Type*} [Ring R] [LinearOrderedCommGroupWithZero Γ₀]
+  [LinearOrderedCommGroupWithZero Γ₀'] {v : Valuation R Γ₀} {w : Valuation R Γ₀'}
+
+/-- If two valuations `v` and `w` are equivalent then `WithVal v` is order-isomorphic
+to `WithVal w`. -/
+def IsEquiv.orderRingIso (h : v.IsEquiv w) :
+    WithVal v ≃+*o WithVal w where
+  __ := equivWithVal v w
+  map_le_map_iff' := by
+    intro a b
+    simp [h.symm (equiv v a), le_def]
+
+@[simp]
+theorem IsEquiv.orderRingIso_apply (h : v.IsEquiv w) (x : WithVal v) :
+    h.orderRingIso x = (equivWithVal v w) x := rfl
```

Comment:
Are we sure that we want this to be the simpNF? I haven't worked with Valued at all but I would suppose that the simpNF _should_ be `(WithVal.equiv v).symm x : WithVal v` (which also suggests that maybe we should have flipped the direction of [WithVal.equiv](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Valued/WithVal.html#WithVal.equiv))